HelgaShiryaeva/assignment1/ReversedListAccessUtil

```diff
@@ -0,0 +1,91 @@
+public class ReversedListAccessUtil {
+    public <T> T getReversed(int index, MyList<T> list) {
+      return list.get(list.size() - index);
+    }
+}
+class MyList<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public MyList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    private void insert(T data, int index) {
+        if (index > size) {
+            throw new IllegalArgumentException("Illegal index : " + index);
+        } else {
+            Node temp = new Node(data);
+            Node current = getNode(index);
+            if (index == 0) {
+                temp.setNext(head);
+                head = temp;
+                tail = head;
+            } else {
+                temp.setNext(current.getNext());
+                current.setNext(temp);
+            }
+            if ( index == size - 1 ) {
+                tail.setNext(temp);
+                tail = temp;
+            }
+        }
+        size++;
+    }
+
+    public void add(T data) {
+        insert(data, size);
+    }
+
+    private Node getNode(int index) {
```

Comment:
`getNode` (and methods that rely on it) behaves in an inconsistent way. Imagine that you have a list of size 1.

- `getNode(0)` will return head (correct).
- `getNode(1)` will also return head (not correct - head is at index 0).